shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/checker/ShardingRuleConfigurationChecker.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.shardingsphere.infra.distsql.exception.resource.RequiredResourceMissedException;
+import org.apache.shardingsphere.infra.distsql.exception.rule.DuplicateRuleException;
+import org.apache.shardingsphere.infra.distsql.exception.rule.InvalidAlgorithmConfigurationException;
+import org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData;
+import org.apache.shardingsphere.infra.rule.identifier.type.DataSourceContainedRule;
+import org.apache.shardingsphere.sharding.api.config.ShardingRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.rule.ShardingAutoTableRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.rule.ShardingTableRuleConfiguration;
+import org.apache.shardingsphere.sharding.spi.KeyGenerateAlgorithm;
+import org.apache.shardingsphere.sharding.spi.ShardingAlgorithm;
+import org.apache.shardingsphere.sharding.support.InlineExpressionParser;
+import org.apache.shardingsphere.spi.typed.TypedSPIRegistry;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Sharding rule configuration checker.
+ */
+public final class ShardingRuleConfigurationChecker {
+    
+    private static final String SHARDING = "sharding";
+    
+    private static final String KEY_GENERATOR = "key generator";
+    
+    private static final Map<String, Class<? extends ShardingSphereAlgorithm>> ALGORITHM_TYPE_MAP = new HashMap<>(2, 1);
+    
+    static {
+        ALGORITHM_TYPE_MAP.put(SHARDING, ShardingAlgorithm.class);
+        ALGORITHM_TYPE_MAP.put(KEY_GENERATOR, KeyGenerateAlgorithm.class);
+    }
+    
+    /**
+     * Check sharding rule configuration.
+     *
+     * @param shardingSphereMetaData ShardingSphere meta data
+     * @param currentRuleConfig      current rule configuration
+     * @throws DistSQLException definition violation exception
+     */
+    public void check(final ShardingSphereMetaData shardingSphereMetaData, final ShardingRuleConfiguration currentRuleConfig) throws DistSQLException {
+        if (null == shardingSphereMetaData || null == currentRuleConfig) {
+            return;
+        }
+        String schemaName = shardingSphereMetaData.getName();
+        checkLogicTables(schemaName, currentRuleConfig);
+        checkResources(schemaName, shardingSphereMetaData, currentRuleConfig);
+        checkKeyGenerators(currentRuleConfig);
+        checkShardingAlgorithms(currentRuleConfig);
+    }
+    
+    private void checkLogicTables(final String schemaName, final ShardingRuleConfiguration currentRuleConfig) throws DistSQLException {
+        LinkedList<String> tablesLogicTables = currentRuleConfig.getTables().stream().map(ShardingTableRuleConfiguration::getLogicTable).collect(Collectors.toCollection(LinkedList::new));
```

Comment:
Can `LinkedList<String>` be changed to `Collection<String>`?

Reply:
ok.